README.md

```diff
@@ -228,6 +228,24 @@ form +++ {
 }
 ```
 
+#### @FormBuilder
+```swift
+@FormBuilder
+var form: Form {
+    nil as Section?
+    Section("Section A") { section in
+        section.tag = "Section_A"
+    }
+    if true {
+        Section("Section B") { section in
+            section.tag = "Section_B"
+        }
+    }
+    nil as TextRow?
```

Comment:
What does this do?

Reply:
Hello @mats-claassen 
Fixed.

Reply:
Hi, so my question is why do we need to put a nil there?

Reply:
What about `if something { nil }` ?

Reply:
Hello @mats-claassen 
OK, I can remove
```
public static func buildOptional(_ components: [RowsProvider]?) -> [BaseRow] {
    components?.flatMap { $0.rows } ?? []
}
```
and
```
public static func buildOptional(_ components: [SectionsProvider]?) -> [Section] {
    components?.flatMap { $0.sections } ?? []
}
```

Reply:
Or should I leave them and remove `if something { nil }` from the examples?

Reply:
I think buildOptional is correct. I just don't think that it is an intuititve example to add `nil` to a form. You could add an optional variable in the example which could be nil but there is no case when you would explicitly put `nil` there, right?

Reply:
Hello @mats-claassen 
Exactly. So, should we remove it from the examples?

Reply:
Or you can add your suggestions to my branch.

Reply:
Hello @mats-claassen 
Can you please tell me what else should I fix in this pull request?

Reply:
Hey @RomanPodymov I think if you remove the `if something { nil}` from the examples then this is fine to be merged

Reply:
Hello @mats-claassen 
Done.